lib/reek/smells/too_many_instance_variables.rb

```diff
@@ -35,14 +35,29 @@ def self.default_config
       #
       def sniff(ctx)
         max_allowed_ivars = value(MAX_ALLOWED_IVARS_KEY, ctx)
-        count = ctx.local_nodes(:ivasgn).map { |ivasgn| ivasgn.children.first }.uniq.length
+
+        count = count_instance_variables(ctx)
+
         return [] if count <= max_allowed_ivars
+
         [smell_warning(
           context: ctx,
           lines: [ctx.exp.line],
           message: "has at least #{count} instance variables",
           parameters: { count: count })]
       end
+
+      private
+
+      def count_instance_variables(ctx)
+        ctx.each_node(:ivasgn, ignored_nodes).map do |node|
```

Comment:
I agree, right now its confusing because of that. Maybe we can revise our approach here and "make it work"?